search/processor.cpp

```diff
@@ -304,15 +304,6 @@ void Processor::SetQuery(string const & query)
   {
     m_tokens.resize(maxTokensCount);
   }
-  else
-  {
-    // Assign the last parsed token to prefix.
-    if (!m_tokens.empty() && !delims(strings::LastUniChar(query)))
-    {
-      m_prefix.swap(m_tokens.back());
-      m_tokens.pop_back();
-    }
-  }
```

Comment:
I have doubts here, because m_prefix is not only for ranking, but for fetching results/suggestions first.
m_tokens work like _full match_, m_prefix works like _prefix match_.

Try "penselv" search query and check the suggestions.

Reply:
Got it. I'll try turning prefixing back on and looking around a little. The prefix token was definitely being ignored in `GetNameScores`, but I can probably fix that less invasively.

I'll go back and take a look at the tests. I was focusing on `search_tests`, but it looks like I broke some of the `search_quality_tests`.

Reply:
Reverted in 1da320 - adding synonym matching to the prefix token fixed the issues I was seeing with st/rd/etc matching without breaking the PREFIX nameScore.